yaml/src/Data/Yaml.hs

```diff
@@ -188,16 +189,22 @@ encodeFile = encodeFileWith defaultEncodeOptions
 -- @since 0.10.2.0
 encodeFileWith :: ToJSON a => EncodeOptions -> FilePath -> a -> IO ()
 encodeFileWith opts fp obj = runConduitRes
-    $ CL.sourceList (objToEvents opts $ toJSON obj)
+    $ CL.sourceList (objToStream opts obj)
    .| Y.encodeFileWith (encodeOptionsFormat opts) fp
 
-objToEvents :: EncodeOptions -> Value -> [Y.Event]
-objToEvents opts o = (:) EventStreamStart
+objToStream :: ToJSON a => EncodeOptions -> a -> [Y.Event]
+objToStream opts o = (:) EventStreamStart
               . (:) EventDocumentStart
-              $ objToEvents' o
+              $ objToEvents opts o
               [ EventDocumentEnd
               , EventStreamEnd
               ]
+
+-- | Encode a value as YAML 'Event's.
```

Comment:
Just one more request on the docs: it's a pretty niche use case to need this function. I'd like to see either (or, ideally, both) of:

1. Including some explanation in the docs here as to why someone may want this
2. Instead of including it in the export list with the more commonly used functions, include it in a separate section for less-commonly-used functions.

Reply:
Maybe it would be better for `objToEvents` (and possibly `objToStream`) to be moved to the `Data.Yaml.Internal` module and exported from there since they're both quite low-level.

There also seems to be another function [`pretty`](https://github.com/snoyberg/yaml/blob/485df7bb08d7ed17a0be363ff493ed27e77fda13/yaml/src/Data/Yaml/Pretty.hs#L68-L79) which is doing a very similar job, but this function is also not exported. Unless I'm mistaken, this version can only print strings containing new lines using single quotes so oddly enough it's less pretty than `objToEvents`.

Is there a reason for both to exist or would you prefer I had a go at refactoring them to use a common function which could be exported from `Data.Yaml.Internal`?

Reply:
Exposing from `.Internal` and refactoring sounds great, assuming no user-visible behavior changes.